website/pages/docs/k8s/installation/multi-cluster/kubernetes.mdx

```diff
@@ -0,0 +1,388 @@
+---
+layout: docs
+page_title: Federation Between Kubernetes Clusters
+sidebar_title: Federation Between Kubernetes Clusters
+description: >-
+  Federating multiple Kubernetes clusters.
+---
+
+# Federation Between Kubernetes Clusters
+
+-> **1.8.0+:** This feature is available in Consul versions 1.8.0 and higher
+
+~> This topic requires familiarity with [Mesh Gateways](/docs/connect/mesh_gateway) and [WAN Federation Via Mesh Gateways](/docs/connect/wan-federation-via-mesh-gateways).
+
+-> Looking for a step-by-step guide? Please follow our Learn Guide: [Secure and Route Service Mesh Communication Across Kubernetes](https://learn.hashicorp.com/consul/kubernetes/mesh-gateways).
+
+This page describes how to federate multiple Kubernetes clusters.
+
+## Primary Datacenter
+
+In order to federate clusters, one cluster must be designated the
+primary datacenter. This cluster will be
```

Comment:
Do you think it's worth a single sentence or so to state that each Kube cluster will be modelled as a separate "Datacenter" regardless of the actual region or zone it's deployed in? Just trying to read this as someone who doesn't know Consul well to start with and the first sentence could be a confusing start?

Reply:
```suggestion
primary datacenter. This datacenter will be
```
"Cluster" is a bit problematic in Consul usage generally but here it could mean "Kube cluster" which isn't wrong but doesn't make as much sense to me when we are talking specifically about what Consul is doing within this cluster and not Kube itself. Not a big deal though if others disagree!

Reply:
> Do you think it's worth a single sentence or so to state that each Kube cluster will be modelled as a separate "Datacenter" regardless of the actual region or zone it's deployed in? Just trying to read this as someone who doesn't know Consul well to start with and the first sentence could be a confusing start?

Added "Consul treats each Kubernetes cluster as a separate Consul datacenter."

> "Cluster" is a bit problematic in Consul usage generally but here it could mean "Kube cluster" which isn't wrong but doesn't make as much sense to me when we are talking specifically about what Consul is doing within this cluster and not Kube itself. Not a big deal though if others disagree!

Totally agree, I've tried to ride the line where if I'm talking about Kubernetes I used cluster and if I'm talking about Consul-specific things I used datacenter. In this case datacenter is the right word in this context and I just missed it.